src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class MethodNameAttribute : Attribute
+    {
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="name">Alternate name of a method.</param>
```

Comment:
Also: what if there are multiple overloads of the same method in the .NET source code, yet they don't all have exactly the same `[MethodName]` attribute argument? What happens then? Can this be documented on this attribute for discoverability?